api/base/serializers.py

```diff
@@ -13,6 +12,27 @@
 from api.base.exceptions import InvalidQueryStringError, Conflict
 
 
+def format_relationship_links(related_link, self_link, rel_meta, self_meta):
+
+    ret = {
+        'links': {
+            'related': {
+                'href': related_link,
+                'meta': rel_meta
+            },
+            'self': {
+                'href': self_link,
+                'meta': self_meta
+            }
+        }
+    }
+
+    # Not currently returning self links
+    if not ret['links']['self']['href']:
```

Comment:
We definitely don't want empty self links, and not just for now. If there aren't self links passed in, they shouldn't be in the structure. I'm just suggesting we do:
```python
if not ret['links']['related']['href']:
    del ret['links']['related']
```
as well. That way, if at some point we have a self link without a related link, there won't be a null related link in there (which would break schema validation). I'm just not sure if that situation is possible.